rtf_vis_tool/src/Components/AllFrustums.js

```diff
@@ -0,0 +1,119 @@
+    */
+    const [frustumsJSX, setFrustumsJSX] = useState([]);
+
+    // Load camera intrinsics and extrinsics by fetching cameras.txt and images.txt respectively.
+    useEffect(() => {
+        
+        // Fetch camera intrinsics from cameras.txt.
+        fetch('results/ba_input/cameras.txt')
+            .then((response) => response.text())
+            .then((in_data) => {
+
+                // Fetch camera extrinsics from images.txt.
+                fetch('results/ba_input/images.txt')
+                .then((response) => response.text())
+                .then((ex_data) => {
+                    visualize_camera_frustums(in_data, ex_data)
+                });
+            });
+    }, []);
+
+    function visualize_camera_frustums(in_data, ex_data) {
+        /* Given all the intrinsic and extrinsic data for all cameras, creates an array of frustums to be 
+        rendered in react three fiber.
+
+        Args:
+            in_data (string): Raw string from cameras.txt, which contains instrinsic parameters of all reconstructed
+                              cameras as per the COLMAP output convention. 
+                              (https://colmap.github.io/format.html#cameras-txt)
+            ex_data (string): Raw string from images.txt, which contains pose and keypoints of all reconstructed images
+                              as per the COLMAP output convention.
+                              (https://colmap.github.io/format.html#images-txt)
+        */
+
+        var in_cameraList = in_data.split('\n');
+        in_cameraList = in_cameraList.slice(3);  // Remove the first 3 lines of comments in cameras.txt.
+        in_cameraList.pop()  // Remove the last empty string from list.
+        
+        var ex_cameraList = ex_data.split('\n');
+        ex_cameraList = ex_cameraList.slice(4); // Remove the 4 lines of comments in images.txt.
+        ex_cameraList.pop() // remove the last empty string from list.
+
+        if (in_cameraList.length != ex_cameraList.length) {
+            alert('Camera count mismatch between images.txt and cameras.txt');
+            return;
+        }
+
+        var finalFrustumsJSX = [];
+        // Combine information from intrinsics and extrinsics to render frustums.
+        for (var i = 0; i < in_cameraList.length; i++) {
+            var inCamArray = in_cameraList[i].split(" ");
+            var exCamArray = ex_cameraList[i].split(" ");
+
+            if (inCamArray[0] !== exCamArray[0]) {
+                alert("Error: Cam IDs in Intrinsic and Extrinsic Files Don't Match.");
+                break;
+            }
+
+            // Set the important variables from inCamArray and exCamArray
+            var fx, img_w, img_h, qw, qx, qy, qz, tx, ty, tz;
+            inCamArray = inCamArray.map(Number);
+            exCamArray = exCamArray.map(Number);
+
+            fx = inCamArray[4];
+            img_w = inCamArray[2];
+            img_h = inCamArray[3];
+            qw = exCamArray[1];
```

Comment:
is it possible to do this on one line in JS, like
```
qw, qx, qy, qz = qw = exCamArray[1], qw = exCamArray[2], qw = exCamArray[3], qw = exCamArray[4];
```

Reply:
maybe that is more readable?

Reply:
yup, I'll make it more compact like that. In JS, the syntax for multi variable assignment on one line looks like this:
![image](https://user-images.githubusercontent.com/40643713/114642610-38696b80-9ca2-11eb-8845-5dda9e8f5eaf.png)